packages/openapi3/test/http-specs/specs.test.ts

```diff
@@ -0,0 +1,164 @@
+import { resolvePath } from "@typespec/compiler";
+import { findTestPackageRoot } from "@typespec/compiler/testing";
+import { deepStrictEqual, ok, strictEqual } from "assert";
+import { exec, execSync } from "child_process";
+import { afterAll, beforeAll, describe, it } from "vitest";
+import { worksFor } from "./../works-for.js";
+import {
+  checkServe,
+  defineSpecTests,
+  markCoverage,
+  validataDataWithSchema,
+} from "./utils/spec-snapshot-testing.js";
+
+const pkgRoot = await findTestPackageRoot(import.meta.url);
+const specsRoot = resolvePath(pkgRoot, "node_modules", "@typespec", "http-specs", "specs");
+const rootOutputDir = resolvePath(pkgRoot, "test", "http-specs", "output");
+
+describe.skip("http-specs convert", () => {
+  defineSpecTests({
+    specDir: specsRoot,
+    outputDir: rootOutputDir,
+    // Excluded cases hits errors/warnings during conversion.
+    exclude: [
+      // warning @typespec/openapi3/invalid-format: Collection format 'tsv' is not supported in OpenAPI3 query parameters. Defaulting to type 'string'.
+      // > 49 |     colors: string[],
+      "parameters/collection-format",
+
+      // warning @typespec/openapi3/xml-unwrapped-invalid-property-type: XML `@unwrapped` can only used on array properties or primitive ones in the OpenAPI 3 emitter, Property 'content' will be ignored.
+      // > 70 |   @unwrapped content: string;
+      "payload/xml",
+
+      // error @typespec/openapi3/unsupported-status-code-range: Status code range '494 to '499' is not supported. OpenAPI 3.0 can only represent range 1XX, 2XX, 3XX, 4XX and 5XX. Example: `@minValue(400) @maxValue(499)` for 4XX.
+      // 59 | model ErrorInRange {
+      "response/status-code-range",
+
+      // error @typespec/openapi3/path-query: OpenAPI does not allow paths containing a query string.
+      // > 435 |       op array(param: string[]): void;
+      "routes",
+    ],
+  });
+});
+
+describe("http-specs cases", () => {
+  worksFor(["3.0.0", "3.1.0"], ({ openApiForFile }) => {
+    describe("AuthApiKeyClient Rest Client", () => {
+      beforeAll(async () => {
+        exec("pnpm spector-serve", { cwd: pkgRoot });
+        await checkServe();
+      });
+
+      afterAll(() => {
+        execSync("pnpm spector-stop", { stdio: "inherit", cwd: pkgRoot });
+      });
+
+      it("should return 204 when the apiKey is valid", async () => {
+        const curr = {
+          name: "authentication/api-key",
+          fullPath: resolvePath(specsRoot, "authentication/api-key"),
+        };
+        const results = await openApiForFile(curr);
+        const res = Object.values(results)[0] as any;
+        const getThing = res.paths["/authentication/api-key/valid"].get;
+
+        const response = await markCoverage(`/authentication/api-key/valid`, {
```

Comment:
this just feels like a lot of duplicate and pointless testing, why are we just redefining some mock calls for the openapi. The goal of this should just be to check we generated it fine.

Not to create a mock client from the openapi and waste time on that.